### PR TITLE
Adding pressure stabilization for single phase poromechanics

### PR DESCRIPTION
**Pull request by ryar9534**
Implemented stabilized flux kernels to enable pressure jump stabilization for single-phase poromechanics. Also moves some stabilization-related functions from MultiphasePoromechanics to PoromechanicsSolver, as now that single-phase stabilization exists it could be unified. 

Also adding input files to perform integrated tests with more physical meaning directly related to pressure stability.

### CODE REVIEW

File: src/coreComponents/physicsSolvers/fluidFlow/SinglePhaseHybridFVM.cpp [hunk 1]
```diff
@@ -260,6 +260,18 @@ void SinglePhaseHybridFVM::assembleFluxTerms( real64 const dt,
...
 
+
+void SinglePhaseHybridFVM::assembleStabilizedFluxTerms( real64 const dt,
+                                                        DomainPartition const & domain,
+                                                        DofManager const & dofManager,
+                                                        CRSMatrixView< real64, globalIndex const > const & localMatrix,
+                                                        arrayView1d< real64 > const & localRhs )
+{
+  // pressure stabilization not implemented
+  return assembleFluxTerms( dt, domain, dofManager, localMatrix, localRhs );
```
Comment (paveltomin) [inline src/coreComponents/physicsSolvers/fluidFlow/SinglePhaseHybridFVM.cpp]: i would avoid doing this and simply crash here

Comment (CusiniM) [reply src/coreComponents/physicsSolvers/fluidFlow/SinglePhaseHybridFVM.cpp]: I agree. Or maybe add a warning and continue? but throwing an error is probably better.

**Review by paveltomin — APPROVED** [review]

File: src/coreComponents/physicsSolvers/fluidFlow/SinglePhaseHybridFVM.cpp (hunk 1 above)
Comment (ryar9534) [reply src/coreComponents/physicsSolvers/fluidFlow/SinglePhaseHybridFVM.cpp]: Good point, I went in and threw errors for the unsupported functions

**Review by CusiniM — APPROVED** [review]
Can we try to use in a case with a fault?

**Comment by ryar9534** [comment]
> Can we try to use in a case with a fault?

We can (it should work?), but I would like to know more about how tpfa is handled to make sure to get the element regions for stabilization right. I also still need to write the integrated tests for this PR.

**Comment by CusiniM** [comment]
> > Can we try to use in a case with a fault?
> 
> We can (it should work?), but I would like to know more about how tpfa is handled to make sure to get the element regions for stabilization right. I also still need to write the integrated tests for this PR.

if we want to add to the frac-matrix and frac-frac connections we probably can. The tpfa is basically the same as for the matrix just with some "special" transmissibilities. It's assembled by looping on the dedicated stencil types.

File: inputFiles/poromechanics/PoroElastic_staircase_co2_3d_base_stab.xml [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+      name="skeletonBarrier"
+      defaultDensity="2650"
+      defaultBulkModulus="5.e9"
+      defaultPoissonRatio="0.25"/>
+    <ConstantPermeability
+      name="rockPermBarrier"
+      permeabilityComponents="{ 0.0, 0.0, 0.0 }"/>
+    <BiotPorosity
+      name="rockPorosityBarrier"
+      grainBulkModulus="1.0e27"
```
Comment (jhuang2601) [inline inputFiles/poromechanics/PoroElastic_staircase_co2_3d_base_stab.xml]:
```suggestion
      defaultGrainBulkModulus="1.0e27"

```

File: inputFiles/poromechanics/PoroElastic_staircase_co2_3d_base_stab.xml [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+      name="skeletonChannel"
+      defaultDensity="2650"
+      defaultBulkModulus="5.e9"
+      defaultPoissonRatio="0.25"/>
+    <ConstantPermeability
+      name="rockPermChannel"
+      permeabilityComponents="{ 9.8e-13, 9.8e-13, 9.8e-13 }"/>
+    <BiotPorosity
+      name="rockPorosityChannel"
+      grainBulkModulus="1.0e27"
```
Comment (jhuang2601) [inline inputFiles/poromechanics/PoroElastic_staircase_co2_3d_base_stab.xml]:
```suggestion
      defaultGrainBulkModulus="1.0e27"

```

File: inputFiles/poromechanics/PoroElastic_staircase_singlephase_3d_base_stab.xml [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+      name="skeletonBarrier"
+      defaultDensity="2650"
+      defaultBulkModulus="5.e9"
+      defaultPoissonRatio="0.25"/>
+    <ConstantPermeability
+      name="rockPermBarrier"
+      permeabilityComponents="{ 0.0, 0.0, 0.0 }"/>
+    <BiotPorosity
+      name="rockPorosityBarrier"
+      grainBulkModulus="1.0e27"
```
Comment (jhuang2601) [inline inputFiles/poromechanics/PoroElastic_staircase_singlephase_3d_base_stab.xml]:
```suggestion
      defaultGrainBulkModulus="1.0e27"
```

File: inputFiles/poromechanics/PoroElastic_staircase_singlephase_3d_base_stab.xml [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+      name="skeletonChannel"
+      defaultDensity="2650"
+      defaultBulkModulus="5.e9"
+      defaultPoissonRatio="0.25"/>
+    <ConstantPermeability
+      name="rockPermChannel"
+      permeabilityComponents="{ 9.8e-13, 9.8e-13, 9.8e-13 }"/>
+    <BiotPorosity
+      name="rockPorosityChannel"
+      grainBulkModulus="1.0e27"
```
Comment (jhuang2601) [inline inputFiles/poromechanics/PoroElastic_staircase_singlephase_3d_base_stab.xml]:
```suggestion
      defaultGrainBulkModulus="1.0e27"
```

**Review by jhuang2601 — APPROVED** [review]

**Comment by paveltomin** [comment]
@ryar9534 there are some conflicts now, please resolve

**Comment by ryar9534** [comment]
> @ryar9534 there are some conflicts now, please resolve

I think I have fixed this

**Pull request merged**